server/models/Availability.js

```diff
@@ -1,59 +1,63 @@
 const mongoose = require("mongoose");
 
-const availabilitySchema = new mongoose.Schema({
-  sitterId: {
-    type: mongoose.Schema.Types.ObjectId,
-    required: true,
-    ref: "User"
+function formatDate(time) {
+  const hour = time.getHours();
+  const min = time.getMinutes();
+  return new Date(0, 0, 0, hour, min, 0);
+}
+
+const availableDaySchema = new mongoose.Schema({
+  start: {
+    type: Date,
+    set: formatDate
   },
-  availableTime: [
-    {
-      day: {
-        type: Date,
-        required: true
-      },
-      start: {
-        type: Date,
-        required: true
-      },
-      end: {
-        type: Date,
-        required: true,
-        validate: {
-          validator: function (endTime) {
-            return endTime > this.start;
-          }
-        }
+  end: {
+    type: Date,
+    set: formatDate,
+    validate: {
+      validator: function (endTime) {
+        return endTime > this.start;
       }
     }
-  ]
+  }
 });
 
-availabilitySchema.virtual("weekDay").get(function () {
-  return this.availableTime.map((item) => {
-    const preFormat = new Date(item.day);
-    const formattedDay = preFormat.toLocaleString(navigator.language, {
-      weekday: "short"
-    });
-    return formattedDay;
-  });
-});
-
-availabilitySchema.virtual("startTime").get(function () {
-  this.availabil;
-  const preFormat = new Date(parseInt(this.availableTime.start));
-  return preFormat.toLocaleTimeString(navigator.language, {
-    hour: "2-digit",
-    minute: "2-digit"
-  });
+const availableWeekSchema = new mongoose.Schema({
+  mon: {
+    availableTime: availableDaySchema
+  },
+  tue: {
+    availableTime: availableDaySchema
+  },
+  wed: {
+    availableTime: availableDaySchema
+  },
+  thu: {
+    availableTime: availableDaySchema
+  },
+  fri: {
+    availableTime: availableDaySchema
+  },
+  sat: {
+    availableTime: availableDaySchema
+  },
+  sun: {
+    availableTime: availableDaySchema
+  }
 });
 
-availabilitySchema.virtual("endTime").get(function () {
-  const preFormat = new Date(parseInt(this.availableTime.end));
-  return preFormat.toLocaleTimeString(navigator.language, {
-    hour: "2-digit",
-    minute: "2-digit"
-  });
+const availabilitySchema = new mongoose.Schema({
+  sitterId: {
+    type: mongoose.Schema.Types.ObjectId,
+    required: true,
+    ref: "User"
+  },
+  isActive: {
```

Comment:
I think we can reference a specific schedule from the Profile model to determine which schedule is active for that particular user?

Reply:
Hi rajiv @rajivtitus ,

Has there been a filed in the Profile model about schedule so far? Do I need to add one?

Reply:
Hi Rajiv @rajivtitus ,

Is there an active schedule field in the Profile model you are referring to? If not, I think I need to add one?

Reply:
> Do you mean we should add a ref in the Profile schema like below:
Yes, that's spot on.

> And leave the isActive field in the availability schema so that we can still set the schedule as active or inactive?
This would hurt the efficiency a bit, would also be redundant: the `Profile` schema can be responsible for tracking which schedule is active, we don't need to also track it on each schedule.